backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/MoveOrCopyDiskCommand.java

```diff
@@ -276,7 +276,7 @@ protected boolean validateSpaceRequirements() {
             } else {
                 getImage().getSnapshots().addAll(diskImageDao.getAllSnapshotsForLeaf(getImage().getImageId()));
             }
-            return validate(storageDomainsValidator.allDomainsHaveSpaceForDisksWithSnapshots(Collections.singletonList(getImage())));
+            return validate(storageDomainsValidator.allTargetDomainsHaveSpaceForDisks(Collections.singletonList(getImage())));
```

Comment:
do we know that this is the cause of exceeding the threshold in the bug? in one of the comments I see they mentioned a simple creation of a preallocated disk - was it mentioned just as a way to reproduce the issue?

Reply:
I originally created this patch to fix copy, and was not aware of the bug. But now that I reread it, I AddDisk does not consider the critical space blocker in the validation. I'll post a separate fix for it, but it seems _slightly_ less severe, as at least we cannot exceed the size like in copy